libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParser.java

```diff
@@ -204,7 +228,11 @@ protected Entitlement parseEntitlement(String scopeName, String entitlementType)
         }
 
         try {
-            return (Entitlement) entitlementConstructor.newInstance(parameterValues);
+            if (entitlementConstructor != null) {
+                return (Entitlement) entitlementConstructor.newInstance(parameterValues);
```

Comment:
Would it be simpler to keep a MethodHandle and invoke that? Then there would be no difference between the ctor and static method?

Reply:
I did a quick prototype of this, but it adds additional complexity for error handling, and doesn't save much in lookup. I'm still open to this, but think I would want to do it as a different PR where we look at simplifying the entirety of the method a bit more.